crates/pathfinder/src/storage/state.rs

```diff
@@ -1773,6 +1775,151 @@ mod tests {
             );
         }
 
+        #[test]
+        fn events_are_ordered() {
+            // This is a regression test where events were incorrectly ordered by transaction hash
+            // instead of transaction index.
+            //
+            // Events should be ordered by block number, transaction index, event index.
+            use crate::core::StarknetTransactionHash;
+            use crate::sequencer::reply::transaction::Event;
+
+            // All events we are storing, arbitrarily use from_address to distinguish them.
+            let expected_events = (0..5)
+                .map(|idx| Event {
+                    data: Vec::new(),
+                    keys: Vec::new(),
+                    from_address: ContractAddress(
+                        StarkHash::from_hex_str(&format!("0x{}", idx)).unwrap(),
```

Comment:
```suggestion
                        StarkHash::from_hex_str(&idx.to_string()).unwrap(),
```

Reply:
Isn't this very contrived any way? 5 is smaller to be it's own hex representation but does it mean we must use it like this? We could do `StarkHash::from_be_slice(&idx.to_be_bytes())`.